scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ServerHandshaker.java

```diff
@@ -470,7 +477,14 @@ private void receivedClientHello(final ClientHello clientHello) throws Handshake
 
 		createServerKeyExchange(clientHello, flight);
 
-		createCertificateRequest(clientHello, flight);
+		boolean clientCertificate = createCertificateRequest(clientHello, flight);
+
+		if (clientCertificate) {
+			states = CLIENT_CERTIFICATE;
+		} else {
+			states = NO_CLIENT_CERTIFICATE;
+		}
+		statesIndex = -1;
```

Comment:
Should we add CLIENT_HELLO to CLIENT_CERTIFICATE and NO_CLIENT_CERTIFICATE initialize `states` with CLIENT_CERTIFICATE (or NO_CLIENT_CERTIFICATE)
And just switch the `states` here if needed ?

Reply:
Though the CLIENT_HELLO is validates ahead, FMPOV, I would not "double check" it here.

Reply:
OK you mean that CLIENT_HELLO is already checked in DTLSConnector ?

I get it, I still feel regarding in term of separation of concerns and readability (avoid the -1 initialization), it could make sense to let ServerHandshaker check CLIENT_HELLO too, but I can understand you want to avoid the double check. :)